java/display-grid/src/main/java/com/esri/arcgisruntime/sample/displaygrid/MainActivity.java

```diff
@@ -0,0 +1,239 @@
+/* Copyright 2018 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.esri.arcgisruntime.sample.displaygrid;
+
+import android.graphics.Color;
+import android.support.v7.app.AlertDialog;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.Button;
+import android.widget.CheckBox;
+import android.widget.Spinner;
+
+import android.widget.Toast;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.geometry.SpatialReference;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.Basemap;
+import com.esri.arcgisruntime.mapping.Viewpoint;
+import com.esri.arcgisruntime.mapping.view.Grid;
+import com.esri.arcgisruntime.mapping.view.LatitudeLongitudeGrid;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.mapping.view.MgrsGrid;
+import com.esri.arcgisruntime.mapping.view.UsngGrid;
+import com.esri.arcgisruntime.mapping.view.UtmGrid;
+import com.esri.arcgisruntime.symbology.LineSymbol;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.symbology.TextSymbol;
+
+public class MainActivity extends AppCompatActivity {
+
+  private MapView mMapView;
+  private CheckBox mLabelsCheckBox;
+  private int mLineColor;
```

Comment:
Hi Trev, the UTM Grid is only visible if you zoom out far out, it is not a problem that's just how it is.
Also, I made those variables global as i use them to save the state of the previous grid when changing the grid. For example, if you change the grid color in lat and long then it will also stay the same color when you switch to UTM or any other. The same applies for label color.  I will try to think over this and see if I can make them local. 
Thanks!

Reply:
Hi Eric! I didn't realize you were calling changeGriColor/changeLabelColor after changing grid type. Global variable absolutely makes sense here!

As to UTM grid not working at those zoom levels. Lets change the zoom level on selection of the grid.